### ISSUE TITLE
Inside FieldGroupListener bean values are not updated

### ISSUE THREAD

**Issue by lightoze** [viritin/viritin#92]
Hello,

I was trying to get notifications about bound bean updates, but when `FieldGroupListener` receives `TextChangeEvent` the bean still contains old values (because `TextChangeEvent` does not change value of the field). As I understood, the whole "eagerValidation" stuff inside e.g. `MTextField` was created to solve this problem.

I have found that attaching the following listener has similar effect, without the need for custom "eagerValidation" logic:
```
public void textChange(FieldEvents.TextChangeEvent event) {
    if (event.getSource() instanceof AbstractTextField) {
        AbstractTextField field = (AbstractTextField) event.getSource();
        field.setValue(event.getText());
        field.setCursorPosition(event.getCursorPosition());
    }
}
```

It has a small downside of cursor flickering when edited text is in the middle of the field, but otherwise work fine. Changes are coming into the bean immediately and validation is performed as soon as the text is typed. Maybe cursor flickering can be also solved by more efficient client-side implementation in Vaadin itself.

Please share your thoughts about this.

**Comment by mstahv** [comment]
Hi, your solution has a nasty side effects when there is a bit more network latency. Typing text into fields will become very tricky. My first try for eager validation looked quite like yours.

Did you try with the last version 1.34? It passes the value from in text change events back to the property (but ignore the value change event it throws back to avoid nasty side effects). I had to do this change for bean level validation, typically cross field. See my [recent blog entry](https://vaadin.com/web/matti/blog/-/blogs/cross-field-validation-in-vaadin).

**Comment by mstahv** [comment]
Hi, did you try with the latest master branch or 1.34? As said, your issue might be already fixed due to the latest cornfield validation changes?

**Comment by lightoze** [comment]
Hi, sorry for a late reply because of vacation. I'm not sure about the version - the last Viritin source code I've checked out to modify is 1.34, but it's possible that I started my experiments with 1.33.

The code I wanted to work was like that:
```
BeanBinder.bind(filter, filterPanel).(group -> System.out.println("Search text: " + filter.getSearch()));
```
And I clearly remember that `filter.getSearch()` always returned the value **before** last field modification. Now it works correctly, so the issue can be closed.
